core/node/proof_data_handler/src/tee_request_processor.rs

```diff
@@ -47,26 +47,62 @@ impl TeeRequestProcessor {
             .await
             .map_err(RequestProcessorError::Dal)?;
 
-        let l1_batch_number_result = connection
-            .tee_proof_generation_dal()
-            .get_next_batch_to_be_proven(request.tee_type, self.config.proof_generation_timeout())
-            .await
-            .map_err(RequestProcessorError::Dal)?;
-
-        let l1_batch_number = match l1_batch_number_result {
-            Some(number) => number,
-            None => return Ok(Json(TeeProofGenerationDataResponse(None))),
-        };
-
-        let tee_verifier_input: TeeVerifierInput = self
-            .blob_store
-            .get(l1_batch_number)
-            .await
-            .map_err(RequestProcessorError::ObjectStore)?;
+        loop {
+            let l1_batch_number = match connection
+                .tee_proof_generation_dal()
+                .get_next_batch_to_be_proven(
+                    request.tee_type,
+                    self.config.proof_generation_timeout(),
+                )
+                .await
+                .map_err(RequestProcessorError::Dal)?
+            {
+                Some(number) => number,
+                None => return Ok(Json(TeeProofGenerationDataResponse(None))),
+            };
+
+            match self.get_blob(l1_batch_number).await {
+                Ok(input) => {
+                    return Ok(Json(TeeProofGenerationDataResponse(Some(Box::new(input)))));
+                }
+                Err(ObjectStoreError::KeyNotFound(_)) => {
+                    tracing::warn!(
+                        "Blob for batch number {} has not been found in the object store. Marking the job as skipped.",
+                        l1_batch_number
+                    );
+                    connection
+                        .tee_proof_generation_dal()
+                        .mark_proof_generation_job_as_skipped(l1_batch_number, request.tee_type)
+                        .await
+                        .map_err(RequestProcessorError::Dal)?;
+                    continue;
+                }
+                Err(err) => return Err(RequestProcessorError::ObjectStore(err)),
+            }
+        }
+    }
 
-        let response = TeeProofGenerationDataResponse(Some(Box::new(tee_verifier_input)));
+    async fn get_blob(
+        &self,
+        l1_batch_number: L1BatchNumber,
+    ) -> Result<TeeVerifierInput, ObjectStoreError> {
+        let max_blob_store_retries = 3;
+        let mut last_err: Option<ObjectStoreError> = None;
+
+        for _ in 0..max_blob_store_retries {
```

Comment:
There's retries logic [embedded](https://github.com/matter-labs/zksync-era/blob/edfcc7dbb7fb60f0f42fff4f3d350974128127b4/core/lib/object_store/src/retries.rs#L27) into `ObjectStore` implementations; is it insufficient? The only way when a retriable error is returned from `get` if these internal retries are exhausted; so wouldn't it be easier to increase the number of retries 3x in the object store config?